### PR TITLE
Minify CSS in style tags

### PR DESCRIPTION
**Pull request by nochso**
- Added CssMinCompressor using natxet/CssMin
- Improved AbstractVendorCompressorTest
  - setup() and tearDown() a protected $compressor for reuse of compressor
- Changed edge-case to reflect change
- Added test for CssMinProcessor
- Modified factories to minify CSS by default

### CODE REVIEW

**Comment by WyriHaximus** [comment]
Thanks, there are two PSR2 errors with your code, I've created a PR to fix them: https://github.com/nochso/HtmlCompress/pull/1

Can you also rebase this PR so I can merge it directly through github?

**Comment by nochso** [comment]
Ok, I merged and force pushed the rebased branch.

I really like [PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) and it's `.php_cs` configuration file. It automatically fixes PSR2 and similar coding style issues with lots of options.

**Comment by nochso** [comment]
Okay, I've got inline CSS working with a basic test.

File: src/Compressor/CssMinCompressor.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+     * Minifies inline CSS attributes.
+     *
+     * @param string $string
+     *
+     * @return string
+     */
+    private function minifyInline($string)
+    {
+        // Get CssMin to compress inline CSS by adding and stripping a selector
+        $mock = 'body{' . $string . '}';
```
Comment (WyriHaximus) [inline src/Compressor/CssMinCompressor.php]: Funny, I've used the exact same hack in the past :+1:

**Comment by WyriHaximus** [comment]
Hmm PHP-CS-Fixer looks interesting, I'll give it a spin.

Everything looks good to me. Anything you like to add before I merge it?

**Comment by nochso** [comment]
I'll have a look at other CSS minifiers and possibly add what I like to `constructSmallest` using a `BestResultCompressor`.

Regarding PHP-CS-Fixer: Here's the [config](https://github.com/nochso/Benchmark/blob/master/.php_cs) I use. It ignores everything from `.gitignore`, adds a copyright header to each file and fixes common PSR2 issues and some style related things. If you want to keep it simple, just use the PSR2 level as the symfony2 level is opinionated.

**Comment by nochso** [comment]
I've added a compressor using https://github.com/websharks/css-minifier

There are also 2 PRs open on that library that will change the outcome of the tests. Either wait for them to be merged or fix the tests when it comes to it?

File: src/Compressor/CssMinifierCompressor.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * CssMinifierCompressor
+ *
+ * @author Marcel Voigt <mv@noch.so>
+ */
+class CssMinifierCompressor extends Compressor
+{
+    protected function execute($string)
+    {
+        return \WebSharks\CssMinifier\Core::compress($string);
```
Comment (WyriHaximus) [inline src/Compressor/CssMinifierCompressor.php]: Can you use a `use` here instead of using the FQCN

File: src/Factory.php [hunk 3]
```diff
@@ -111,12 +119,23 @@ public static function constructSmallest()
...
+                    [
+                        'patterns' => [
+                            Patterns::MATCH_STYLE,
+                            Patterns::MATCH_STYLE_INLINE,
+                        ],
+                        'compressor' => new BestResultCompressor(
+                            [
+                                new CssMinCompressor(),
+                                new CssMinifierCompressor(),
+                            ]
```
Comment (WyriHaximus) [inline src/Factory.php]: Can you add the `ReturnCompressor` to this list just in case

**Comment by WyriHaximus** [comment]
Thanks I'll try it out soon :+1: 

We have two options, merge without those PR's and tag `1.2` and then tag `1.2.1` when those PR's are merged. Or wait for them and include them in `1.2`. You're call, both are fine with me.

**Comment by nochso** [comment]
I've amended the last commit with the `use` and `ReturnCompressor` changes. Make sure to update.

Rather not wait, who knows when they'll react. Plus it's frozen by composer.lock :)

**Comment by WyriHaximus** [comment]
It's frozen for this project but this packages `composer.lock` doesn't affect projects requiring it. BUT due to their versioning system it is frozen in `composer.json` :+1: .

Anyway, I'm merging this and tagging `1.2.0`. Thank you for your effort :+1: :D

**Comment by nochso** [comment]
Great. Thank you too :)

**Comment by WyriHaximus** [comment]
Shoot me a new PR when those others are merged and tagged 👍
